tests/features/steps/artifacts-poetry-plugin-steps.py

```diff
@@ -15,52 +13,107 @@
+    def __init__(self, name, pretty_version, files) -> None:
         self.name = name
         self.pretty_version = pretty_version
+        self.files = files
 
     def name(self):
         return self.name
 
     def pretty_version(self):
         return self.pretty_version
 
+    def files(self):
+        return self.files
+
 
 @given("the following artifacts")
 def the_following_artifacts(context):
     context.packages = {}
     for row in context.table:
+        name = row["package"]
+        version = row["version"]
+        filename = f"{name}-{version}.whl"
+        file = open(os.path.join(TEST_RESOURCES_DIR, filename), "w")
+        file.close()
+        files = []
+        files.append(filename)
         context.packages[row["key"]] = Package(
-            name=row["package"], pretty_version=row["version"]
+            name=name, pretty_version=version, files=files
         )
 
 
-@given("a Python project with dependencies on artifacts {keys}")
+@given("a Python project with dependencies on package {keys}")
 def step_impl(context, keys):
-    shutil.copy(TEMPLATE_PY_PROJECT, PY_PROJECT)
-    with open(PY_PROJECT, "a") as py_project:
-        py_project.write("\n[tool.poetry.dependencies]\n")
+    with open(POETRY_LOCK, "w") as poetry_lock:
+        poetry_lock.write("\n[metadata]\n")
+        poetry_lock.write('lock-version = "2.0"\n')
+        poetry_lock.write('python-versions = "^3.8"\n\n')
         for key in keys.split(","):
             package = context.packages[key]
-            py_project.write(f'{package.name} = "{package.pretty_version}"\n')
-        py_project.write(f'python = "^3.8"')
-    subprocess.run(["poetry", "install", "--no-root"], cwd=TEST_RESOURCES_DIR)
+            files = package.files
+            poetry_lock.write("[[package]]\n")
+            poetry_lock.write(f'name = "{package.name}"\n')
+            poetry_lock.write(f'version = "{package.pretty_version}"\n')
+            poetry_lock.write(f'description = "Test"\n')
+            poetry_lock.write(f"optional = false\n")
+            poetry_lock.write(f'python-versions = ">=2.6, !=3.0.*, !=3.1.*, !=3.2.*"\n')
+            poetry_lock.write(f"files = [\n")
+            for file in files:
+                poetry_lock.write(
+                    "{"
+                    + f'file = "{file}", hash = "sha256:ebda1a6c9e5bfe95c5f9f0a2794e01c7098b3dde86c10a95d8621c5907ff6f1c"'
+                    + "},\n"
+                )
+            poetry_lock.write("]\n\n")
 
 
-@when("the Python dependency search is triggered")
+@when("the Python deployable dependency is triggered.")
 def step_impl(context):
     poetry = Factory().create_poetry(Path(POETRY_LOCK))
-    context.response_packages = getPackages(poetry)
+    deploy_packages, non_deploy_packages = get_deploy_packages(
+        poetry, TEST_RESOURCES_DIR
+    )
+    context.deploy_packages = deploy_packages
+    context.non_deploy_packages = non_deploy_packages
+
 
+def get_package_set(packages):
```

Comment:
Q: Is this just to strip out duplicates or is something more interesting happening here?